x-pack/platform/plugins/shared/integration_assistant/public/components/create_integration/create_integration_assistant/flyout/cel_configuration/steps/upload_spec_step/api_definition_input.tsx

```diff
@@ -86,28 +93,43 @@ export const ApiDefinitionInput = React.memo<ApiDefinitionInputProps>(
 
           if (fileContent == null) {
             setApiFileError(i18n.API_DEFINITION_ERROR.CAN_NOT_READ);
+            onModifySpecFile(false);
             return;
           }
 
           if (fileContent === '' && e.loaded > 100000) {
             // V8-based browsers can't handle large files and return an empty string
             // instead of an error; see https://stackoverflow.com/a/61316641
             setApiFileError(i18n.API_DEFINITION_ERROR.TOO_LARGE_TO_PARSE);
+            onModifySpecFile(false);
             return;
           }
 
           const prepareResult = prepareOas(fileContent);
 
           if ('error' in prepareResult) {
             setApiFileError(prepareResult.error);
+            onModifySpecFile(false);
             return;
           }
 
           const { oas } = prepareResult;
 
+          const oasPaths = oas.getPaths();
+
+          // Verify we have valid GET paths in the uploaded spec file
+          if (
+            Object.keys(oasPaths).length === 0 ||
+            Object.values(oasPaths).filter((path) => path?.get).length === 0
```

Comment:
The first condition is redundant, isn't it?
```suggestion
            Object.values(oasPaths).filter((path) => path?.get).length === 0
```